src/App.tsx

```diff
@@ -1,61 +1,128 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const defaultId = '0';
+
+export const App: React.FC = () => {
+  const [choosedUserId, setChoosedUserId] = useState(defaultId);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [visibleTodos, addVisibleTodo] = useState(todos);
+  const [isSubmitted, setSubmit] = useState(false);
+
+  const handleSelectingUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setChoosedUserId(event.target.value);
+  };
+
+  const handleEnteringTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTodoTitle(event.target.value);
+  };
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const title = todoTitle.trim();
+
+    const id = visibleTodos
+      .reduce((maxId, todo) => (maxId > todo.id ? maxId : todo.id), 0) + 1;
+    const user = getUser(Number(choosedUserId));
+    const userId = user && user.id;
+    const completed = false;
+
+    if (title === '' || choosedUserId === '0' || !user || !userId) {
```

Comment:
Why you don’t use the default variable?
```suggestion
    if (title === '' || choosedUserId === DEFAULT_USER_ID || !user || !userId) {
```